modules/customer/includes/commerce_customer.entity.inc

```diff
@@ -25,7 +26,7 @@ class CommerceCustomerProfile extends Entity {
 
   public function uri() {
     if (empty($this->id)) {
-      return NULL;
+      return array();
```

Comment:
@argiepiano as I was updating the commerce entities to include the new requirement of a "path" array I Just set these all to return null in these cases (where they wouldn't have their own uri without an id available, I think?) to match the spec from core:

https://github.com/backdrop/backdrop/blob/c53e13d829406bcfcae365754d8345dd57da69f3/core/modules/entity/entity.class.inc#L69

I don't have a problem with an array here really, but is the spec wrong or am I confused?

Reply:
I believe those docs are incorrect. If you look at [this line](https://github.com/backdrop/backdrop/blob/c53e13d829406bcfcae365754d8345dd57da69f3/core/modules/entity/entity.theme.inc#L22) in `template_preprocess_entity()`, returning `NULL` will generate a fatal error, since it uses the `+` operator for arrays.

Reply:
I'll open an issue in the main issue queue.

Reply:
grrr @ docs! Okay sounds good I will change the rest of these back as well.

Reply:
@elisseck I've submitted an issue and PR for the documentation, in case you can review it and mark it as WFM . It's just a simple change in the docs. https://github.com/backdrop/backdrop-issues/issues/6749

Reply:
The rest of these were swapped back in 9c2a0c2a514286c5d441beb3b87d437b143a51e1